File: C++/0027.cpp
#include <vector>

class Solution {
public:
    int removeElement(std::vector<int>& nums, int val) {
        int idx = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != val) {
                nums[idx++] = nums[i];
            }
        }
        return idx;
    }
};

// Topic: Array
// Time Complexity: O(n)
// Space Complexity: O(1)

File: Python/0027.py
from typing import List

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        pointer = 0
        for i in range(len(nums)):
            if nums[i] != val:
                nums[pointer] = nums[i]
                pointer += 1
        
        return pointer

# Topic: Array
# Time complexity: O(n)
# Space complexity: O(1)